### PR TITLE
[Feature] starlet support starcachelib

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
add some configs in be so that cloud table can use starcache

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -823,6 +823,10 @@ CONF_Int32(starlet_cache_dir_allocate_policy, "0");
 // Buffer size in starlet fs buffer stream, size <= 0 means not use buffer stream.
 // Only support in S3/HDFS currently.
 CONF_Int32(starlet_fs_stream_buffer_size_bytes, "131072");
+CONF_mBool(starlet_use_star_cache, "false");
+CONF_mString(starlet_star_cache_mem_size_limit, "4%");
```
Comment (kevincai) [inline be/src/common/config.h:827]: completely disable mem usage?

Comment (starrocks-xupeng) [reply be/src/common/config.h:827]: no, starcache's meta needs this, just like tablet meta cache

Comment (kevincai) [reply be/src/common/config.h:827]: this is strange. Assume the meta may require 100MB mem, when I set to less than 100MB, the starcache can't even work? or will just use about let's say 70MB, and then out of space. It leaves the question to the user, how much mem I have to set if I want to give about X MB to cache, do  I need to provide X / (1 - y%) overhead or not? what's the overhead of y%?

Comment (starrocks-xupeng) [reply be/src/common/config.h:827]: will default to 0 when next starlet version is ready

**Review by kevincai — APPROVED** [review]

**Review by sduzh — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**